completion/test_utils.py

```diff
@@ -65,7 +65,9 @@ def override_waffle_switch(self, override):
         Parameters:
             override (bool): True if tracking should be enabled.
         """
-        _waffle_overrider = waffle.waffle().override(waffle.ENABLE_COMPLETION_TRACKING, override)
+        _waffle_overrider = override_waffle_switch(
+            waffle.ENABLE_COMPLETION_TRACKING_SWITCH, override
+        )
         _waffle_overrider.__enter__()
         self.addCleanup(_waffle_overrider.__exit__, None, None, None)
```

Comment:
Did you mean to delete this whole method?  It seems like you replaced it with setup/teardown.

Reply:
No, the completion mixin has two mechanisms to override the waffle switch:

1. The first is the setUpClass/tearDownClass trick by which the override switch default value is set to the class' COMPLETION_SWITCH_ENABLED attribute.
2. The second is this decorator, which overrides the default value set by the setup/tearDown hack.

Reply:
I somewhat see. :) I find it confusing that we have two functions named `override_waffle_switch`, and I'm not clear on where the class level one is called.  Could you rename the class level `override_waffle_switch` to `override_completion_tracking_switch`?

Reply:
There is just one `override_waffle_switch`, which is the one from edx-toggles . The other one is called `override_completion_switch`: https://github.com/edx/completion/pull/118/files/67f276a0231e80ebbb6009151cf50d0d3edf381a#diff-f47a4a90118f43738723e016b4338cce98e746a466d978c89127abab6c66b5eaR114

The setUpClass/tearDownClass methods are automatically called by unittest on test class creation: https://docs.python.org/3.7/library/unittest.html#unittest.TestCase.setUpClass

Reply:
That's not what I see in this file: https://github.com/edx/completion/blob/67f276a0231e80ebbb6009151cf50d0d3edf381a/completion/test_utils.py

Reply:
Right! So that makes three overriding mechanisms...

Reply:
Unfortunately, I cannot rename `CompletionWaffleTestMixin.override_waffle_switch` because this method is used in many places in edx-platform... :sweat:

Reply:
Got it.  Options include:
1. Ignore and do nothing.
2. Adjust the import to provide a second name.
3. Introduce a new name for the one in the class, and keep the older version and mark it as deprecated.  We could then move to the newer name at some point in the future.
I'll leave this to you to decide.

Reply:
I'm going to go with option 1...